Adding .github repository

egekorkan opened the pull request:
As we have spoken on our friday calls, we decided to have a repository for project-level "stuff", which are:
- GitHub discussions for the whole project, mainly organizational
- GitHub organization landing page so that the project has a similar landing to https://github.com/nodejs
- Mirroring project description so that we can version it

egekorkan commented:
@netomi let's wait for @relu91 and @danielpeintner to have a look first before merging

relu91 commented on otterdog/eclipse-thingweb.jsonnet:
```diff
@@ -17,6 +17,20 @@ orgs.newOrg('eclipse-thingweb') {
+    orgs.newRepo('.github') {
+      allow_update_branch: false,
+      default_branch: "main",
+      dependabot_security_updates_enabled: true,
```
In my understanding, this is a no-code repository, right? Do we need dependabot?

egekorkan replied:
Yup only readme at best. Removing

netomi commented on otterdog/eclipse-thingweb.jsonnet:
```diff
@@ -17,6 +17,20 @@ orgs.newOrg('eclipse-thingweb') {
   _repositories+:: [
+    orgs.newRepo('.github') {
+      allow_update_branch: false,
+      default_branch: "main",
```
you can remove the two settings here:

- default_branch
- dependabot_security_updates_enabled

as the same values are already inherited from the default config, see its values here

https://eclipse-thingweb.github.io/.eclipsefdn/

You should see that also in the comment added to the PR from the validation workflow.

egekorkan replied:
I have removed them and thanks for the pointer to the website, it helps actually quite a lot :)

netomi commented:
If you want to enable organization wide discusssions, you will have to set has_discussions to true for organization settings and specify discussion_source_repository to the .github repo.

danielpeintner commented:
Proposed PR looks good to me with the changes/updates proposed by @netomi (hence I wait before approving...)

egekorkan commented:
I have incorporated the changes and did the other changes:
- Added description which is our new "teaser" text at https://projects.eclipse.org/projects/iot.thingweb
- Added thingweb emailing list as email

danielpeintner reviewed (approved):
LGTM

netomi commented:
the change has been applied.

Fyi: changes have to be applied manually for now after a PR has been merged. It uses some command line tool to do the change, but it still requires someone to manually run the tool.

In your case, I have run into some small issues when trying to apply the changes, thats why it took a bit of time. The changes are now live. For the organization description: I have realized now that GitHub has a maximum of 160 chars for the description, so I shortened the one you provided.

For the  discussion_source_repository, you need to specify the full slug including the owner like that: "eclipse-thingweb/.github"

I have added this validation rules in the meantime to the tool so that we can avoid that confusion in the future.

egekorkan commented:
Ok I see, I thought it was automated but this is also fine :)

The description looks fine, thanks!
